boards/stk3700/Makefile.include

```diff
@@ -8,5 +8,8 @@ include $(RIOTMAKE)/tools/serial.inc.mk
 # setup JLink for flashing
 JLINK_PRE_FLASH = r
 
+# setup emulator
+include $(RIOTMAKE)/tools/renode.inc.mk
```

Comment:
This could be included globally in `$(RIOTBASE)/Makefile.include` if the `emulate` target is part of the `MAKECMDGOALS` variable. This would avoid the need to duplicate this line in all boards with the emulate support.

```diff
diff --git a/Makefile.include b/Makefile.include
index 350d15fd08..bc749ffe8a 100644
--- a/Makefile.include
+++ b/Makefile.include
@@ -389,6 +389,11 @@ else
   include $(RIOTMAKE)/dependency_resolution.inc.mk
 endif
 
+# Include renode emulator variables if the emulate target is used
+ifneq (,$(filter emulate,$(MAKECMDGOALS)))
+  include $(RIOTMAKE)/tools/renode.inc.mk
+endif
+
 # Include Board and CPU configuration
 INCLUDES += $(addprefix -I,$(wildcard $(BOARDDIR)/include))
 include $(BOARDDIR)/Makefile.include
```

It might even be possible to check if the board provides the renode feature at that level.

Reply:
Renode is an implementation of `make emulate`. There could be others (currently none). How would that work in that case?

Reply:
Also, users could potentially define a different `RENODE_BOARD_CONFIG`. How would that work with this approach? Similar to options like `JLINK_PRE_FLASH`, etc.

Reply:
> How would that work in that case?

Let's say you can add `EMULATOR ?= renode` in the board `Makefile.include` and move the logic above after including this Makefile. (so after including `include $(BOARDDIR)/Makefile.include`).
We could do the same for programmers and serial by the way but that's out of scope here.

Reply:
Doesn't [microbit](https://github.com/RIOT-OS/RIOT/blob/master/boards/microbit/Makefile.include) use `qemu-system-arm` as an emulator?